main_test.go

```diff
@@ -356,6 +362,9 @@ func Test_prepareServerOptions(t *testing.T) {
 }
 
 func Test_preparedServer(t *testing.T) {
+	// reset global http server
+	http.DefaultServeMux = new(http.ServeMux)
```

Comment:
No, please don't do modify global state for this. Just inject the handlers instead.

Reply:
(By "inject", I mean "pass them as an argument")